tree/ntuple/v7/src/RField.cxx

```diff
@@ -207,6 +207,18 @@ std::tuple<const void *const *, const std::int32_t *, const std::int32_t *> GetR
    return {GetRVecDataMembers(const_cast<void *>(rvecPtr))};
 }
 
+/// Applies the field IDs from 'from' to 'to', where from and to are expected to be each other's clones.
+/// Used in RClassField and RCollectionClassField cloning. In these classes, we don't clone the subfields
+/// but we recreate them. Therefore, the on-disk IDs need to be fixed up.
+void SyncFieldIDs(const ROOT::Experimental::Detail::RFieldBase &from, ROOT::Experimental::Detail::RFieldBase &to)
+{
+   auto iFrom = from.cbegin();
+   auto iTo = to.begin();
+   for (; iFrom != from.cend(); ++iFrom, ++iTo) {
+      iTo->SetOnDiskId(iFrom->GetOnDiskId());
+   }
```

Comment:
Compared to the previous version in this PR, the `RSchemaIteratorTemplate<bool>` only visits descendant fields.  Instead, I think it should be
```suggestion
   to.SetOnDiskId(from.GetOnDiskId());
   auto iTo = to.begin();
   for (auto iFrom = from.cbegin(); iFrom != from.cend(); ++iFrom, ++iTo) {
      iTo->SetOnDiskId(iFrom->GetOnDiskId());
   }
```

Reply:
It's true that we only visit descendant fields but that should be enough. The ID of the mother field itself is set by `Clone()`.

Reply:
Oh, that's absolutely right!